### PR TITLE
Changelog page that pulls from Github Release notes

### LINKED ISSUE #831: Build Changelog that pulls from Github Release notes
## Feature Overview

This feature pulls the most recent Github release into the app to show what's new.

Image

## Requirements

_If there is a missing / incorrect requirement, please leave a comment before starting work on this._

- [ ] What's new page pulls the latest Github release with notes

## Implementation Suggestions

_Below are some ideas for implementation to get you started.  Use your best judgment here—if there's a better way to do things, go for it!_

- We already have some Github API related code written that might be helpful here:

https://github.com/maybe-finance/maybe/blob/main/app/models/provider/github.rb

## Designs

Below are the designs you should follow while implementing this:

https://www.figma.com/design/lonJmVk3HYkwZoIO7xYP2w/Maybe-App-(Community)?node-id=3789-202&t=Zy48LHlrHgM2ZErq-0

## Reminders

- Make sure to review our [contributing guidelines](https://github.com/maybe-finance/maybe/blob/main/CONTRIBUTING.md) before starting on an issue
- We do our best to define a clear spec for new features and fixes, but think of them as "suggestions", not "hard requirements".  We welcome ideas and suggestions!
    - If you see missing requirements to this issue, please leave a comment below explaining what is missing and why it is important.
    - If you see a requirement that you think is _incorrect_ or _not optimal_, please leave a comment explaining what you think needs to change below.

### PR DESCRIPTION
**Pull request by mattia-malnis**
Closes #831

https://github.com/maybe-finance/maybe/assets/15941778/153a4235-357b-4f81-9801-639d2d736131

I'm not 100% sure that I wrote the CSS correctly using Tailwind. Any suggestions are welcome!

I also think it could be a good idea to rename "changelog" to "what's new" in the menu that opens when clicking the avatar image.

immagine

### CODE REVIEW

File: app/assets/stylesheets/application.tailwind.css [hunk 1]
```diff
@@ -81,6 +81,43 @@
...
+  }
+  .releases-notes-details summary {
+    @apply flex items-center gap-x-1;
+  }
+  .releases-notes-details .dropdown-caret {
+    @apply content-none border-4 border-b-0 border-transparent border-t-gray-500 size-0 inline-block;
+  }
+  .releases-notes-details .octicon {
+    @apply inline-block overflow-visible align-text-bottom fill-current;
+  }
```
Comment (zachgoll) [inline app/assets/stylesheets/application.tailwind.css:120]: Definitely a tricky one here.  I think this is a good opportunity for us to evaluate the usage of typography within the app.

Ideally, I think what we should probably consider here is an API like so:

```erb
<div class="prose prose--github-release-notes">
  <%= release_notes[:body].html_safe %>
</div>
```

We have `prose` as the "base" styling mechanism, and then `prose--github-release-notes` as a "specialization" where we can add styles for things like `.user-mention`, `.octicon`, and `.dropdown-caret` which don't exist anywhere else.

That way, in `tailwind.config.js`, we can do all of our "base" [prose overrides](https://github.com/tailwindlabs/tailwindcss-typography?tab=readme-ov-file#customizing-the-css):

```js
// SAMPLE, not actual styles

/** @type {import('tailwindcss').Config} */
module.exports = {
  theme: {
    extend: {
      typography: {
        DEFAULT: {
          css: {
            color: '#333',
            a: {
              color: '#3182ce',
              '&:hover': {
                color: '#2c5282',
              },
            },
          },
        },
      },
    },
  },
  plugins: [
    require('@tailwindcss/typography'),
    // ...
  ],
}
```

And then in our global Tailwind CSS file, we can apply the "specializations":

```css
.prose--github-release-notes {
  .octicon {
    @apply inline-block overflow-visible align-text-bottom fill-current;
  }

  .dropdown-caret {
    @apply content-none border-4 border-b-0 border-transparent border-t-gray-500 size-0 inline-block;
  }
}
```

So in summary, the changes required here would be:

1. Delete existing `.prose` class in Tailwind CSS file (it's not being used anywhere)
2. Add `.prose--github-release-notes` and all the "specializations" to the Tailwind CSS file
3. Apply all "base" prose overrides directly in `tailwind.config.js`
4. Move all non-typography styles directly to the ERB files (e.g. the `.release-notes-row`, etc.)

Let me know what you think!

File: app/views/pages/changelog.html.erb [hunk 2]
```diff
@@ -4,9 +4,25 @@
...
+              <%= image_tag release_notes[:avatar], class: "rounded-full w-full h-full object-cover" %>
+            </div>
+          <div>
+            <div class="text-gray-900 font-medium text-sm"><%= release_notes[:name] %></div>
+            <div class="text-gray-500 text-sm"><%= release_notes[:published_at].strftime("%B %d, %Y") %></div>
+          </div>
+        </div>
+        </div>
+        <div class="w-2/3 releases-notes-details">
+          <h1 class="mb-5 text-xl text-gray-900"><%= release_notes[:name] %></h1>
```
Comment (zachgoll) [inline app/views/pages/changelog.html.erb:21]: Small nitpick here, but we should probably change to an `h2` to avoid duplicate H1s on the page.

**Review by zachgoll — COMMENTED** [review]
Looks great, thanks for tackling this!

Visually it all looks good.  I left some suggestions around how we can consolidate some of the "prose" styles and leverage the Tailwind typography plugin a bit.

File: app/assets/stylesheets/application.tailwind.css (hunk 1 above)
Comment (zachgoll) [reply app/assets/stylesheets/application.tailwind.css:120]: @mattia-malnis for those "base" prose styles for the Tailwind Config file, I worked with @justinfar to standardize this a bit and included it in the updated design file:

https://www.figma.com/design/lonJmVk3HYkwZoIO7xYP2w/Maybe-App-(Community)?node-id=3789-202&t=UY4qtUbF3hRvnpp1-0

Please note—this is an _example_ that we should try to be as "generic" as possible with when applying to those "base" styles in the Tailwind config.  These "base" styles should work with _any_ HTML we need to render.

**Comment by flacnut** [comment]
Thanks for doing this!

**Comment by mattia-malnis** [comment]
@zachgoll thanks for the feedback! I think I can finish it tomorrow!

File: app/assets/stylesheets/application.tailwind.css (hunk 1 above)
Comment (mattia-malnis) [reply app/assets/stylesheets/application.tailwind.css:75]: @zachgoll your solution is definitely better, I didn't know about this possibility with Tailwind! I reviewed the code, let me know if everything is ok

File: app/views/pages/changelog.html.erb (hunk 2 above)
Comment (mattia-malnis) [reply app/views/pages/changelog.html.erb]: I've changed it to an h2 to avoid duplicate H1s on the page!

File: app/views/pages/changelog.html.erb [hunk 3]
```diff
@@ -4,9 +4,25 @@
 <div class="space-y-4">
   <h1 class="text-gray-900 text-xl font-medium mb-4">What's New</h1>
```
Comment (zachgoll) [inline app/views/pages/changelog.html.erb:5]: I know other spots in the codebase are still lacking this, but we should probably include an i18n `t(".title")` here

**Review by zachgoll — APPROVED** [review]
Awesome work, this looks great!

File: app/views/pages/changelog.html.erb (hunk 3 above)
Comment (mattia-malnis) [reply app/views/pages/changelog.html.erb]: Sure, I updated it!
